test/MaticTokenVesting.test.js

```diff
@@ -0,0 +1,147 @@
+      let balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), "0")
+      let amount = toWei("1900000000")
+      await vesting.release(1)
+      balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), amount.toString())
+      amount = toWei((117990560 + 1900000000).toString())
+      await vesting.release(2)
+      balance = await token.balanceOf.call(accounts[0])
+      assert.equal(balance.toString(), amount.toString())
+    })
+
+    it("should test token vesting for userX", async function() {
+      const amount = toWei("10")
+      let block = await web3.eth.getBlock("latest")
+      let blockTime = block.timestamp
+      let time = new Date(blockTime)
+      time.setMinutes(time.getMinutes() + 6)
+      time = +time
+      await vesting.addVesting(accounts[1], time.toString(), amount, {
+        from: accounts[0]
+      })
+      await token.transfer(vesting.address, amount)
+      const balance = await vesting.vestingAmount(28)
+      assert.equal(balance.toString(), amount.toString())
+    })
+
+    it("Removing a vesting entry with the owner account", async function() {
+      let result = await vesting.removeVesting(3, { from: owner })
+      await vesting.retrieveExcessTokens(result.receipt.logs[0].args["2"], {
+        from: owner
+      })
+    })
+
+    it("Removing a vesting entry with a non-owner account", async function() {
+      await assertRevert(vesting.removeVesting(3, { from: accounts[1] }))
+    })
+
+    it("Trying to remove an unexistent vesting entry", async function() {
+      await assertRevert(vesting.removeVesting(30, { from: owner }))
+    })
+
+    it("Trying to remove an already released vesting entry", async function() {
+      await assertRevert(vesting.release(1, { from: owner }))
+    })
+
+    it("Trying to remove an already removed vesting entry", async function() {
+      await assertRevert(vesting.removeVesting(30, { from: owner }))
+    })
+
+    it("Trying to add a vesting entry from a non-owner account", async function() {
+      const amount = toWei("10")
+      let block = await web3.eth.getBlock("latest")
+      let blockTime = block.timestamp
+      let time = new Date(blockTime)
+      time.setMinutes(time.getMinutes() + 6)
+      time = +time
+      await assertRevert(
+        vesting.addVesting(accounts[1], "" + time, amount, {
+          from: accounts[1]
+        })
+      )
+    })
+
+    it("Trying to release the tokens associated with existing vesting entry", async function() {
+      let amount = await token.balanceOf(vesting.address)
+      await assertRevert(vesting.retrieveExcessTokens(amount, { from: owner }))
+    })
+
+    it("should test token vesting for userX1", async function() {
+      let seconds = 60 * 6000
+      await increaseBlockTime(seconds)
+      await mineOneBlock()
+      const amount = toWei("10")
+      await vesting.release(28)
+      const balance = await token.balanceOf.call(accounts[1])
```

Comment:
You'd need to assert that balanceOf(user) before releasing was 0.

Reply:
yeah that we can do, but we know it's fresh state and never used that account 😛.

Reply:
@0xAshish That's how you write test case 🤪